lib/puppet/util/logging.rb

```diff
@@ -281,8 +281,8 @@ def issue_deprecation_warning(message, key, file, line, use_caller)
         $deprecation_warnings[key] = message
         call_trace = use_caller ?
           (offender || get_deprecation_offender).join('; ') :
-          "#{file || 'unknown'}:#{line || 'unknown'}"
-        warning("#{message}\n   (at #{call_trace})")
+          "#{file || _('unknown')}:#{line || _('unknown')}"
```

Comment:
Sounds good. Only the "unknown" string and final message at the bottom need to be extracted then. The middle man is all formatted variables. Updated.